### PR TITLE
Fix `self` return typehint being generated with a leading slash

### PR DESCRIPTION
**Pull request by kieranajp**
Hi :wave: 

First, thanks for making my favourite test framework :blush: 

Second, I've come across a small bug this afternoon where, when trying to create a double for an interface with a `self` return typehint, `\self` was generated instead.

Quick example - with the following interface:

```php
interface Message
{
    public function setId(MessageId $id): self;
}
```

And the following test code:

```php
$message = Double::instance([
    'implements' => Message::class,
]);
```

This resulted in:

```
PHP Fatal error:  '\self' is an invalid class name in /server/http/vendor/kahlan/kahlan/src/Plugin/Double.php(104) : eval()'d code on line 7
```

This PR is just a quick fix to get past this.

Let me know if there's anything more you need from me on this!

### CODE REVIEW

File: src/Plugin/Double.php
Comment (samsonasik) [inline src/Plugin/Double.php:408]: need to have strict check : 

```suggestion
        if (in_array($trimmed, [ 'self', 'static' ], true)) {
```

**Comment by kieranajp** [comment]
Thanks for the super fast review! 

I'm not sure how to go about getting the tests passing here: 

```
actual:
        (string) "<?php\nnamespace Kahlan\\Spec\\Plugin\\Double;\n\nclass Double490 implements \\Kahlan\\Spec\\Mock\\Plugin\\Double\\HelloInterface {\n\n    public function hello() : self {}\n    public function aloha() : \\Kahlan\\Spec\\Mock\\Plugin\\Double\\HelloInterface {}\n\n}\n?>"
      expected:
        (string) "<?php\nnamespace Kahlan\\Spec\\Plugin\\Double;\n\nclass Double0 implements \\Kahlan\\Spec\\Mock\\Plugin\\Double\\HelloInterface {\n\n    public function hello() : self {}\n    public function aloha() : \\Kahlan\\Spec\\Mock\\Plugin\\Double\\HelloInterface {}\n\n}\n?>"
```

Running it with focus mode seems to generate the `Double0` class name; am I missing a trick for not having to append the number in my test case?

**Comment by samsonasik** [comment]
@kieranajp thank you, wait for @jails approval ;)

**Review by samsonasik — APPROVED** [review]

**Comment by jails** [comment]
Thanks for the PR, I'll need to do some cleanup first and I'll push a new release right after.

**Comment by kieranajp** [comment]
Awesome, thank you! :)

**Comment by jails** [comment]
Fixed in 5.0.7

**Pull request merged**
